### ISSUE TITLE
consider using Square's OkHttp library

### ISSUE THREAD

**Issue by gesellix** [gesellix/docker-client#26]
see http://square.github.io/okhttp/

- [x] can replace our own plain HttpClient
- [x] supports unix sockets (Unix, Docker for Mac)
- [x] supports named pipes (Windows)
- [x] supports WebSockets
- [x] few dependencies
- [ ] could be useful to implement interactive Docker `attach` or `exec`

**Comment by gesellix** [comment]
meh. OkHttp doesn't allow `unix` schemes. Only `http`/`https`.

**Comment by gesellix** [comment]
Feedback wanted:

The [OkHttp branch](https://github.com/gesellix/docker-client/tree/OkHttp) contains a refactored internal http client. Instead of using the plain Java api (HttpUrlConnection, etc) it now delegates to Square's [OkHttp](https://github.com/square/okhttp) library.

I would like to know whether I forgot some details or if the refactoring breaks something. So, on top of the unit and integration tests, I'd like **you** to help me finding issues. The check is actually quite simple, given the following setup:

- you need to be able to clone this repository
- you need to have Java installed

In summary, these are the steps you would need to perform:

````
git clone https://github.com/gesellix/docker-client
cd docker-client
checkout OkHttp
./gradlew checkDockerClient
````

Every possible platform should work, e.g.:

- plain simple Linux
- OS X via Docker Machine or boot2docker
- OS X via Docker for Mac (beta)
- Windows via Docker Machine or boot2docker
- Windows via Docker for Windows (beta)
- anything else? Please tell me!

**Comment by gesellix** [comment]
OkHttp is now used internally, now we still need to implement interactive attach/exec. Will be continued in #30.
